flang/test/Lower/associate.f90

```diff
@@ -0,0 +1,27 @@
+! RUN: bbc %s -o - | tco | llc --relocation-model=pic --filetype=obj -o %assoc.o
+! RUN: %CC %assoc.o -L%L -lFortranRuntime -lFortranDecimal -lstdc++ -lm -o %assoc.exe
+! RUN: %assoc.exe | FileCheck %s
+
+program p
+  integer :: n, foo
+  n = 0
+  associate (i => n, j => n + 10, k => foo(20))
+!   CHECK: 0 0 10 20
+    print*, n, i, j, k
+    n = n + 1
+!   CHECK: 1 1 10 20
+    print*, n, i, j, k
+    i = i + 1
+!   CHECK: 2 2 10 20
+    print*, n, i, j, k
+  end associate
+! CHECK: 2
+  print*, n
+end
+
+integer function foo(x)
+  integer x
+  integer, save :: i = 0
+  i = i + x
+  foo = i
+end function foo
```

Comment:
A small regression test, not necessarily end-to-end, would also be nice for the do concurrent locality part of the PR (I guess not in this test file though).

Reply:
do concurrent code has been removed